akvo/rsr/static/tests/scripts-src/my-results/utils.test.js

```diff
@@ -9,14 +9,191 @@
+    it('expect to return true if value is numeric', () => {
+        const number = 55;
+        expect(utils.isNumeric(number)).to.be.true;
+    });
+    it('expect to return false if values not numeric', () => {
+        const string = 'noNumeric';
+        expect(utils.isNumeric(string)).to.be.false;
+    });
+});
+
+describe("function idsToActiveKey", () => {
+    it('expect to return IDs as an array of strings', () => {
+        const ids = [1, 2, 3, 4];
+        expect(utils.idsToActiveKey(ids)).to.satisfy(function(array) {
+            return array.every(function(item) {
+                return typeof item === 'string';
+            })
+        });
+    });
+});
+
+describe("arrowFunction isNewUpdate", () => {
+    it('expect to return true if the update is new', () => {
+        const newUpdate = {id: 'new-', name: 'update'};
+        expect(utils.isNewUpdate(newUpdate)).to.be.true;
+    });
+    it('expect to return true if we pas a string starting with "new-"', () => {
+        const newString = "new-17";
+        expect(utils.isNewUpdate(newString)).to.be.true;
+    });
+    it('expect to return false if the update is not new', () => {
+        const notNewUpdate = {id: '4', name: 'update'};
+        expect(utils.isNewUpdate(notNewUpdate)).to.be.false;
+    });
+});
+
+describe("function collapseId", () => {
+    it('expect to create an collapsedId from the model name and parent object id', () => {
+        const model = 'model2',
+              id = 200;
+        expect(utils.collapseId(model, id)).to.equal('model2-200');
+    });
+});
+
+// requires that the function tested is changed to an exported function
+describe("function childModelName", () => {
+    it('expect to return child model name', () => {
+        const model = c.OBJECTS_RESULTS,
+              child = c.OBJECTS_INDICATORS;
+        expect(utils.childModelName(model)).to.equal(child);
+    });
+});
+
+describe("function parentModelName", () => {
+    it('expect to return parent model name', () => {
+        const model = c.OBJECTS_INDICATORS,
+              parent = c.OBJECTS_RESULTS;
+        expect(utils.parentModelName(model)).to.equal(parent);
+    });
+});
+
+describe("function levelAbove", () => {
+    it('expect to compare index of array to decide that level is above', () => {
+        const models = [1, 2],
+              model = models[1],
+              compare = models[0];
+        expect(utils.levelAbove(model, compare)).to.equal(model < compare);
+    });
+});
+
+describe("function levelBelow", () => {
+    it('expect to compare index of array to decide that level is below', () => {
+        const models = [1, 2],
+              model = models[0],
+              compare = models[1];
+        expect(utils.levelBelow(model, compare)).to.equal(model > compare);
+    });
+});
+
+// requires that the function tested is changed to an exported function
+describe("function flatten", () => {
+    it('expect to return an flatten array of arrays', () => {
```

Comment:
I'd rather describe it as 'expect to return an flattened array of a nested original'